pkg/watermark/fetch/offset_timeline.go

```diff
@@ -170,18 +175,14 @@ func (t *OffsetTimeline) GetEventTimeFromInt64(inputOffsetInt64 int64) int64 {
 	t.lock.RLock()
 	defer t.lock.RUnlock()
 
-	var (
-		offset    int64 = -1
-		eventTime int64 = -1
-	)
+	var eventTime int64 = -1
 
 	for e := t.watermarks.Front(); e != nil; e = e.Next() {
 		// get the event time has the closest offset to the input offset
 		// exclude the same offset because this offset may not finish processing yet
-		// offset < e.Value.(OffsetWatermark).offset: use < because we want the largest possible timestamp
-		if offset < e.Value.(OffsetWatermark).offset && e.Value.(OffsetWatermark).offset < inputOffsetInt64 {
-			offset = e.Value.(OffsetWatermark).offset
+		if e.Value.(OffsetWatermark).offset < inputOffsetInt64 {
```

Comment:
why is this condition changed?

Reply:
it's the same logic but a cleaner implementation:
because OT is ordered from highest to smallest, so as soon as we find one offset that is smaller than the input offset, we can return it